bn_mp_radix_size.c

```diff
@@ -23,42 +22,66 @@ mp_err mp_radix_size(const mp_int *a, int radix, int *size)
       return MP_OKAY;
    }
 
-   /* special case for binary */
-   if (radix == 2) {
-      *size = mp_count_bits(a) + ((a->sign == MP_NEG) ? 1 : 0) + 1;
+   /* A small shortcut for powers of two. */
```

Comment:
This looks kind of ugly. Could you check for power of two instead via bitwise and then shift bit_count in a for loop iterating over the radix zero bits?

Reply:
> This looks kind of ugly.

Beauty lies in the eyes of the beholder! ;-)

What do you mean by "bitwise"? Something like `(x != 0) && (!(x&(x-1)))`? And then count the zero LSB's?
Can do, if you insist?

Reply:
Yes, I think I would like that more.

Reply:
Well, now it's me who finds it a bit ugly. ;-)

Reply:
Puuh but you could make it better. Why is rem a toplevel variable. Why not use it directly in the condition? Radix should probably also be unsigned in the arguments. Since we are deprecating the old function we can also change that?

Reply:
I like it more since it looks more general, no hardcoded lut.

Reply:
> Why is rem a toplevel variable. Why not use it directly in the condition?

It was a simple C&P because I wasn't sure if that's exactly what you wanted. If you want it that way I can put a small speck of Vienna chalk on it and buff it a little.

Reply:
Whatever :)